"gmake install" fails on FreeBSD 12.2

FreeBSD is not one of the platforms for which the `zq` tools are subjected to automated testing and release packaging. However, since it's all written in Go, we expect it would compile and run. A community user recently expressed an interest in perhaps starting to run `zq` on FreeBSD, so I spun up a VM and checked to see if "just works".

Alas, as of `zq` commit `10804d2` on FreeBSD 12.2, it did fail. Not sure what to make of it.

```
$ gmake install
...
# github.com/brimsec/zq/pkg/rlimit
pkg/rlimit/raiseopenfileslimit_unix.go:16:18: cannot assign uint64 to rlimit.Cur (type int64) in multiple assignment
pkg/rlimit/raiseopenfileslimit_unix.go:16:39: cannot use rlimit.Max (type int64) as type uint64 in argument to kernMaxFiles
pkg/rlimit/raiseopenfileslimit_unix.go:26:15: cannot use rlimit.Cur (type int64) as type uint64 in return argument
gmake: *** [Makefile:143: install] Error 2
```

--- a/pkg/rlimit/kernmaxfiles_darwin.go
+++ b/pkg/rlimit/kernmaxfiles_darwin.go
@@ -2,17 +2,18 @@ package rlimit
 
 import (
 	"fmt"
+	"syscall"
 
 	"golang.org/x/sys/unix"
 )
 
-func kernMaxFiles(max uint64) (uint64, error) {
+func kernMaxFiles(rlimit *syscall.Rlimit) error {
 	kernMax, err := unix.SysctlUint32("kern.maxfilesperproc")
 	if err != nil {
-		return 0, fmt.Errorf("systcl: %w", err)
+		return fmt.Errorf("systcl: %w", err)
 	}
-	if uint64(kernMax) < max {
-		return uint64(kernMax), nil
+	if uint64(kernMax) < rlimit.Max {
+		rlimit.Max = uint64(kernMax)
 	}
-	return max, nil
+	return nil
 }

--- a/pkg/rlimit/kernmaxfiles_notdarwin.go
+++ b/pkg/rlimit/kernmaxfiles_notdarwin.go
@@ -1,7 +1,0 @@
-// +build !darwin
-
-package rlimit
-
-func kernMaxFiles(max uint64) (uint64, error) {
-	return max, nil
-}

--- a/pkg/rlimit/kernmaxfiles_unixnotdarwin.go
+++ b/pkg/rlimit/kernmaxfiles_unixnotdarwin.go
@@ -1,0 +1,9 @@
+// +build !darwin,!plan9,!windows
+
+package rlimit
+
+import "syscall"
+
+func kernMaxFiles(rlimit *syscall.Rlimit) error {
+	return nil
+}

--- a/pkg/rlimit/raiseopenfileslimit_unix.go
+++ b/pkg/rlimit/raiseopenfileslimit_unix.go
@@ -12,16 +12,15 @@ func raiseOpenFilesLimit() (uint64, error) {
 	if err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, &rlimit); err != nil {
 		return 0, fmt.Errorf("getrlimit: %w", err)
 	}
-	var err error
-	rlimit.Cur, err = kernMaxFiles(rlimit.Max)
-	if err != nil {
+	if err := kernMaxFiles(&rlimit); err != nil {
 		return 0, err
 	}
+	rlimit.Cur = rlimit.Max
 	if err := syscall.Setrlimit(syscall.RLIMIT_NOFILE, &rlimit); err != nil {
 		return 0, fmt.Errorf("setrlimit: %w", err)
 	}
 	if err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, &rlimit); err != nil {
 		return 0, fmt.Errorf("getrlimit: %w", err)
 	}
-	return rlimit.Cur, nil
+	return uint64(rlimit.Cur), nil
 }

